vmdb/app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -586,11 +589,9 @@ def prov_req_submit
   def prov_get_form_vars
     if params[:ids_checked]                         # User checked/unchecked a tree node
       ids = params[:ids_checked].split(",")
-      fld = @edit[:wf].kind_of?(MiqHostProvisionWorkflow) ? "tag_ids" : "vm_tags"
-      @edit[:new]["#{fld}".to_sym] = Array.new
-      ids.each do |id|
-        @edit[:new]["#{fld}".to_sym].push(id.to_i) if id != "" #for some reason if tree is not expanded clicking on radiobuttons this.getAllChecked() sends up extra blanks
-      end
+      @edit.store_path(:new, tag_symbol_for_workflow, [])
+      # for some reason if tree is not expanded clicking on radiobuttons this.getAllChecked() sends up extra blanks
+      @edit[:new][tag_symbol_for_workflow].push(ids.collect { |i| i.to_i unless i.blank? }.compact)
```

Comment:
@kbrock Good catch, thanks!  I implemented your suggestions of ```&:present?``` and ```&:to_i```.  Also, looking over it again, we're storing a blank array two lines above, so i just combined it into one line to make it simpler.